### PR TITLE
Apply ldapUserFilter on members of group

### PR DESCRIPTION
**Pull request by Cybso**
Refers to issue #8220

user_ldap configured with custom filters for active directory access
(group-member-association is "member"). Then it can happen that the
a group contain members that don't belong to the users
available in Nextcloud (the most trivial reason is that the user filter
contains "(!(UserAccountControl:1.2.840.113556.1.4.803:=2))" to exclude
disabled users from being imported).

This can be fixed by applying the ldapUserFilter when resolving the UID
for a DN fetched from the group's member list.

Signed-off-by: Roland Tapken <roland@bitarbeiter.net>

### CODE REVIEW

File: apps/user_ldap/lib/Access.php [hunk 1]
```diff
@@ -564,7 +564,11 @@ public function dn2ocname($fdn, $ldapName = null, $isUser = true, &$newlyMapped
 		}
 
 		if(is_null($ldapName)) {
-			$ldapName = $this->readAttribute($fdn, $nameAttribute);
+			if ($isUser) {
+				$ldapName = $this->readAttribute($fdn, $nameAttribute, $this->connection->ldapUserFilter);
+			} else {
+				$ldapName = $this->readAttribute($fdn, $nameAttribute);
```
Comment (blizzz) [inline apps/user_ldap/lib/Access.php]: for consistency, applying group filter would also make sense in this case. A $filter var can be created in the beginning of the function, so an additional if won't be necessary

File: apps/user_ldap/lib/Access.php [hunk 2]
```diff
@@ -564,7 +564,11 @@ public function dn2ocname($fdn, $ldapName = null, $isUser = true, &$newlyMapped
...
+			}
 			if(!isset($ldapName[0]) && empty($ldapName[0])) {
 				\OCP\Util::writeLog('user_ldap', 'No or empty name for '.$fdn.'.', \OCP\Util::INFO);
```
Comment (blizzz) [inline apps/user_ldap/lib/Access.php]: The message needs to be adjusted and take this case into account, too.

**Comment by Cybso** [comment]
@blizzz I've applied the requested changes, please review again

**Review by blizzz — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
🐘

**Comment by MorrisJobke** [comment]
CI failure is unrelated.

**Comment by zeridon** [comment]
Unfortunately this does not fix all possible cases.

Currently (nc 13.0.2 with this patch) when trying to view the group that has a disabled user in it the spinner spins and spins and never finishes.

The error reported in the logs is: 

```
OC\User\NoUserException: REDACTED is not a valid user anymore
[internal function] OCA\User_LDAP\User_LDAP->getHome('REDACTED')
/var/www/html/apps/user_ldap/lib/User_Proxy.php - line 108: call_user_func_array(Array, Array)
/var/www/html/apps/user_ldap/lib/Proxy.php - line 150: OCA\User_LDAP\User_Proxy->callOnLastSeenOn('REDACTED', 'getHome', Array, false)
/var/www/html/apps/user_ldap/lib/User_Proxy.php - line 227: OCA\User_LDAP\Proxy->handleRequest('REDACTED', 'getHome', Array)
/var/www/html/lib/private/User/User.php - line 282: OCA\User_LDAP\User_Proxy->getHome('REDACTED')
/var/www/html/settings/Controller/UsersController.php - line 261: OC\User\User->getHome()
/var/www/html/settings/Controller/UsersController.php - line 322: OC\Settings\Controller\UsersController->formatUserForIndex(Object(OC\User\User))
[internal function] OC\Settings\Controller\UsersController->index(0, 50, 'Law', '', '')
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 161: call_user_func_array(Array, Array)
/var/www/html/lib/private/AppFramework/Http/Dispatcher.php - line 91: OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Settings\Controller\UsersController), 'index')
/var/www/html/lib/private/AppFramework/App.php - line 115: OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Settings\Controller\UsersController), 'index')
/var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OC\\Settings\\Con...', 'index', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/var/www/html/lib/private/Route/Router.php - line 297: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/var/www/html/lib/base.php - line 999: OC\Route\Router->match('/settings/users...')
/var/www/html/index.php - line 37: OC handleRequest()
{main}
```

**Comment by blizzz** [comment]
@zeridon right, this was not backported 13. Mostly because we tagged it as enhancement, in fact it fixes a bug. Thanks for pointing it out! Next  window is 13.0.5.

**Pull request merged**